### ISSUE TITLE
[Feature]: Edit a playwright script via codegen

### ISSUE THREAD

**Issue by IgorGee** [microsoft/playwright#7515]
### Feature request

I have an existing playwright script, but I want to take some additional complicated steps in the middle of that existing script.

As far as I understand, I have to manually write those steps or `codegen` from scratch.
I'd like to use the same great `codegen` tool to edit my existing script.

Some ideas of how to implement this:

* `codegen --edit existing-script.js --line 46`
* `codegen --edit existing-script.js`
  * (this would work by having some identifier to replace in the file. `// {{ insert point }}`

**Comment by dgozman** [comment]
@IgorGee You can use `page.pause()` to run codegen from an arbitrary place. Here is the [documentation](https://playwright.dev/docs/cli#codegen-with-custom-setup). Does that work for you?

**Comment by IgorGee** [comment]
@dgozman Hmm, it's close, but the behavior of codegen at the pause point is not behaving the same way as it would if I created a brand new script.

What I'm experiencing:
1. I create a pause point on `existing-script.js`
2. Run `node existing-script.js`
3. Playwright gets to the pause point correctly, but it's only recording `page.goto` commands during the pause.

It's not recording arbitrary clicks or interactions within a page.

I can't tell if this is a bug or intended behavior.

**Comment by dgozman** [comment]
@IgorGee There is a "Record" button at the top-left that toggles recording mode. It is not toggled by default when using `page.pause()`, only when started with `npx playwright codegen`, so you'll have to click it manually before recording actions.

**Comment by IgorGee** [comment]
@dgozman Yep, "Record" was on after the pause. I didn't mention that in my previous message, but it was definitely on and still exhibits the behavior I described.

**Comment by pavelfeldman** [comment]
I just tried it and it did work for me:
- I ran the script, it paused,
- I hit Record, it focused target page and created a blank recording snippet
- I started performing actions and they were recorded

Could you share a minimal repro from your side?

**Comment by IgorGee** [comment]
@pavelfeldman I just tried to reproduce using google as an example site, and was unable to.

Seems like the bug is specific to the site I was trying to automate. I made a dummy account to reproduce the bug. The bug will only happen after a login. My guess is that playwright gets confused with some kind of redirect from the URL.

Here's the repo to reproduce:
https://github.com/IgorGee/playwright-repro-pause
